src/internal/connector/graph_connector.go

```diff
@@ -233,6 +234,10 @@ type getOwnerIDAndNamer interface {
 // api to fetch the user or site using the owner value. This fallback assumes
 // that the owner is a well formed ID or display name of appropriate design
 // (PrincipalName for users, WebURL for sites).
+//
+// Consumers are allowed to pass in a path suffix (eg: /sites/foo) as a site
+// owner, but only if they also pass in a nameToID map.  A nil map will cascade
+// to the fallback, which will fail for having a malformed id value.
```

Comment:
I have a few concerns with where/how this is implemented and the comments
* it's not clear from the comment alone how much the suffix encompasses as it never states what the SP URL is like to begin with. For example, is `foo` a valid suffix?
* it's not clear this will work in all cases (maybe partly due to the above) as it's not clear if it will allows matching on multiple sites. There's no logic to select the longest match so final results may actually be indeterminate. For example, if the suffix is `foo`, what if there is `sites/foo` and `sites/bar/foo` (idk if `sites/bar/foo` is valid as I don't know as much about SP but seems like we should have answers to this if we want to add this feature or we need more definition as to what a suffix is)?
* I'm hesitant to implement this in GraphConnector's init logic (which iirc is where `getOwnerIDAndNameFrom` is called). Waiting until this point makes it difficult to enforce invariants about the state of data which can make reasoning about things difficult. I would suggest that if we want to allow this we require that some higher layer fixup stuff such that we have a well-formed, fully qualified URL by the time we reach this function (and an instance of GC being created for use during backup). This also avoids weirdness about the fallback failing and generally makes things more consistent

Reply:
1: updated the comment for clarity.  2: going to check in with georgi on multiple matches.  My assumption is that we'll want to include all; but I hadn't thought of that up front, so it might require a second pass at the logic here.  3:

> init logic

Nitpicky: this isn't init so much as optional validation/population.  You can use GC without performing these calls.  Still, per location, I think this is the appropriate place for implementation.  After all, the intent of this logic is to validate and enforce expected preconditions.  Not to consume other layer's enforcement of them.

> higher layer fixup stuff such that we have a well-formed, fully qualified URL by the time we reach this function

The core problem here is that we can't construct a fully qualified url without knowing the hostname of the tenant.  [To do that](https://goodworkaround.com/2022/02/09/finding-azure-ad-domain-from-tenant-id/) not only requires a graph call, but a new permission we aren't currently requiring (might need it in the future, though, so I'll look into that as a request).  I see that as an iteration we can make, more than a requirement for this change.

Reply:
> >init logic
>
>Nitpicky: this isn't init so much as optional validation/population. You can use GC without performing these calls. Still, per location, I think this is the appropriate place for implementation. After all, the intent of this logic is to validate and enforce expected preconditions. Not to consume other layer's enforcement of them.

My concern is a bit more general than just the function I commented on. I'm more concerned with the entire initialization flow. Sure, the individual function may not be called, but if I were try to write down the state that GC ended up in after initialization I would have trouble doing that, and that complexity invites bugs because it's difficult to reason what state things are in and/or miss considering a case. If I were to take a stab at it, it would maybe be something like (including this change)
* populates name and ID from IDNamer if given
* if IDNamer not given, attempts to get it from Graph
* may throw an error, but only if IDNamer wasn't given and Graph didn't have it (for non-prefix users and sites)
* may *not* throw an error if IDNamer *was* given and a site prefix *was* given
* may throw an error if IDNamer *wasn't* given and a site prefix *was* given

Reply:
Well good news!  We're dropping suffix matching support for v0.  I'm going to close this PR and make a cleanup in its place.